modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotValidator.scala

```diff
@@ -132,29 +133,60 @@ object CurrencySnapshotValidator {
         }
       })
 
-      val recreateFn = (trigger: ConsensusTrigger) =>
+      def recreateFn(trigger: ConsensusTrigger) =
         mkEvents.flatMap { events =>
-          currencySnapshotCreator
-            .createProposalArtifact(lastArtifact.ordinal, lastArtifact, lastContext, trigger, events, rewards, facilitators)
+          def usingKryo =
```

Comment:
I would consider

```scala
          def usingHasher = (hasher: Hasher[F]) => currencySnapshotCreator
            .createProposalArtifact(
              lastArtifact.ordinal,
              lastArtifact,
              lastContext,
              hasher,
              trigger,
              events,
              rewards,
              facilitators
            )
          
          def usingKryo = usingHasher(Hasher.forKryo)

          def usingJson = usingHasher(Hasher.forJson)
```

because the only difference between these 2 is the hasher provided but it is still not very DRY

Reply:
:+1: